NMS-13339: Add FtpDetector docs (#3446)
OpenNMS/opennms: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc

= FTP Service Detector

> The FtpDetector is able to validate ftp connection dial-up processes.

Reviewer: ```suggestion
This detector lets you find FTP services for monitoring in your network.
```

> The detector can test ftp server on multiple ports and specific login data.

Reviewer: ```suggestion
It finds FTP server on multiple ports and can use also login credentials for service detection.
```

Reviewer: We detect a FTP service using the command channel (21/tcp), right?

Author: Yes. https://github.com/OpenNMS/opennms/blob/7e9f5759badaf5636ae487a72b605a2a89e3e13e/opennms-provision/opennms-detector-lineoriented/src/main/java/org/opennms/netmgt/provision/detector/simple/FtpDetector.java#L46

== Detector facts

[options="autowidth"]
|===
| Implementation | `org.opennms.netmgt.provision.detector.simple.FtpDetector`
|===

== Configuration and Use

.Parameters for the FTP Service Detector
[options="header, %autowidth"]
|===
| Parameter            | Description   | Required | Default value
| `port`               | Port to query | optional | `21`

> | `multilineIndicator` | ???           | optional | `-`

Reviewer: ???

Author: Feel free to add a useful description :-) I am not sure what exactly this is doing. The [monitor](https://github.com/OpenNMS/opennms/blob/develop/docs/modules/operation/pages/service-assurance/monitors/FtpMonitor.adoc) does not have this parameter.

Author: ```suggestion
```

|===

== Example Configuration

[source,xml]
----
<detector name="FTP" class="org.opennms.netmgt.provision.detector.simple.FtpDetector">
  <parameter key="port" value="21"/>

>   <parameter key="multilineIndicator" value="-"/>

Reviewer: Maybe adding an example with credentials?

Author: I thought this is not possible due to the missing parameters here:
![image](https://user-images.githubusercontent.com/4995222/120555152-eec41400-c3fa-11eb-8e03-0b1bee70be97.png)

Author: ```suggestion
```

= FTP Service Detector

The FtpDetector is able to validate ftp connection dial-up processes.

> The detector can test ftp server on multiple ports and specific login data.

Reviewer 2: ```suggestion
It finds the FTP server on multiple ports and can also use login credentials for service detection.
```
